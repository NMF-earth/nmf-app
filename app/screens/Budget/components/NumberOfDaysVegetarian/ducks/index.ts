import selectors from "./NumberOfDaysVegetarian.selectors";

export { selectors };
