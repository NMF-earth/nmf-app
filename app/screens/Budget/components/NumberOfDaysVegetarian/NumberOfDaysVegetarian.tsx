import React from "react";
import { View } from "react-native";
import { useSelector } from "react-redux";
import { isEmpty, pipe, not } from "ramda";
import { Text } from "../../../../components";
import styles from "./NumberOfDaysVegetarian.styles";
import { selectors } from "./ducks";

const NumberOfDaysVegetarian = () => {
  const isAnyMeatEmissionSaved = useSelector(selectors.isAnyMeatEmissionSaved);
  const daysWithoutEatingMeat = useSelector(selectors.getDaysWithoutEatingMeat);

  let days = null;
  let message = "";

  if (daysWithoutEatingMeat < 1) {
    message = "Meat is murder, please don't eat animals 🥺";
  } else if (daysWithoutEatingMeat < 8) {
    days = daysWithoutEatingMeat + " days";
    message = "That's a nice start! 👌";
  } else {
    days = daysWithoutEatingMeat + " days";
    message = "Well done! 🎉";
  }

  if (!isAnyMeatEmissionSaved) {
    message =
      "Seems like you haven't eaten any meat since you are using the app, good job! 🍾";
  }

  return (
    <View style={styles.container}>
      {isAnyMeatEmissionSaved && days ? (
        <View style={styles.daysContainer}>
          <Text.H3 green>{days}</Text.H3>
          <Text.H3> without meat!</Text.H3>
        </View>
      ) : null}
      <Text.H3 center>{message}</Text.H3>
    </View>
  );
};

export default NumberOfDaysVegetarian;
