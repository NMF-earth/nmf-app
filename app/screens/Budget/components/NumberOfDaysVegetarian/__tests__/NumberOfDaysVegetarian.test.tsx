import React from "react";
import { create } from "react-test-renderer";

import NumberOfDaysVegetarian from "../NumberOfDaysVegetarian";

jest.unmock("../NumberOfDaysVegetarian");

it("NumberOfDaysVegetarian renders correctly", () => {
  const tree = create(<NumberOfDaysVegetarian />).toJSON();
  expect(tree).toMatchSnapshot();
});
