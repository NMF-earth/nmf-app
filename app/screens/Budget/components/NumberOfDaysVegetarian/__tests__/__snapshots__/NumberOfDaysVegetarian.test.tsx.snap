// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NumberOfDaysVegetarian renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "marginBottom": 26,
      "marginTop": 10,
    }
  }
>
  <View
    style={
      Object {
        "flex": 1,
        "flexDirection": "row",
      }
    }
  >
    <Text.H3
      green={true}
    >
      function mockConstructor() {
        return fn.apply(this, arguments);
      } days
    </Text.H3>
    <Text.H3>
       without meat!
    </Text.H3>
  </View>
  <Text.H3
    center={true}
  >
    Well done! ðŸŽ‰
  </Text.H3>
</View>
`;
